### DISCUSSION TITLE
Negative emittance value

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#775] [Python AT]
Hi,

I encountered an issue recently that AT returns a negative value for the vertical emittance for FCC-ee  lattice with corrected alignments errors. Here is the code I used for the calculation:

r = at.Lattice(ring, energy=45.6e9)
r.harmonic_number = 135000
r.set_cavity(Voltage=200e6)
r.enable_6d(dipole_pass='BndMPoleSymplectic4RadPass',
            quadrupole_pass='StrMPoleSymplectic4RadPass',
            sextupole_pass='StrMPoleSymplectic4RadPass')
r.set_rf_frequency()
r.set_cavity_phase()
params = r.envelope_parameters()
emittance_h = params.emittances[0]
emittance_v = params.emittances[1]
print('Horizontal emittance:', emittance_h * 1e9, 'nm')
print('Vertical emittance:', emittance_v * 1e12, 'pm')

Horizontal emittance: 0.0003683509681623736 
Vertical emittance: -5.022314775696341e-07

Any insights or suggestions  would be appreciated.

**Comment by oscarxblanco** [comment]
Dear @elafmusa ,
could you try `emit0,_,_ = ring.ohmi_envelope()` ?
 I had better results with it because it takes into account the lattice coupling.

**Reply by elafmusa** [reply]
Dear @oscarxblanco 
Thank you for your response, i have tried to use ohmi_envelope() as you suggested, however, the obtained vatical emittance still of negative value.

**Comment by swhite2401** [comment]
If I understand well you get -5e-19, this can probably be assumed as being zero... however, it would be very surprising to have zero residual for a lattice with errors even after corrections so something is off.

What are the values expected for the ideal lattice? Are the damping times ok? Are you sure your lattice is stable? These would be the first things to check if ohmi_envelope() gives non-physical results

**Reply by elafmusa** [reply]
Dear @swhite2401 , Thank you for your valuable insight! checking the radiation parameters, the lattice was clearly unstable, the issue was fixed by setting proper RF voltage. Best
